### ISSUE TITLE
[Bug]: 1.0.0-alpha.1 JavaScript parsing error: Expression expected

### ISSUE THREAD

**Issue by escaton** [web-infra-dev/rspack#7049]
### System Info

System:
    OS: macOS 14.2.1
    CPU: (12) arm64 Apple M2 Pro
    Memory: 5.50 GB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.11.0 - ~/.nvm/versions/node/v20.11.0/bin/node
    Yarn: 4.1.0 - ~/.nvm/versions/node/v20.11.0/bin/yarn
    npm: 10.2.4 - ~/.nvm/versions/node/v20.11.0/bin/npm
  Browsers:
    Chrome: 126.0.6478.127
    Safari: 17.2.1
  npmPackages:
    @rspack/cli: 1.0.0-alpha.1 => 1.0.0-alpha.1
    @rspack/core: 1.0.0-alpha.1 => 1.0.0-alpha.1
    @rspack/plugin-react-refresh: 1.0.0-alpha.1 => 1.0.0-alpha.1

### Details

After updating from `1.0.0-alpha.0` to `1.0.0-alpha.1` got this error on production build:
```
ERROR in × JavaScript parsing error: Expression expected

 24522 │             ERRORS_SENT_COUNT++, // noop
 24523 │             ;(0,ErrorTracker_/* getErrorTracker */.K)().debug(Error('***'));
       ·             ─
```

here is the related piece of source:
```
				if (ERRORS_SENT_COUNT++ < 3) {
					// noop
				}

				getErrorTracker().debug(new Error('***'))
```

The comma appears regardless of what is under the condition. It seems like the problem is the increment operator. I got 3  errors like this and in all cases, there is an increment in the condition.

`git bisect` pointed to this [PR](https://github.com/web-infra-dev/rspack/pull/7028) @ahabhgk FYI

### Reproduce Steps

Run production build with rspack

**Comment by markmssd** [comment]
@escaton I'm having the same issue, upgrading from `1.0.0-alpha.0` to `1.0.0-alpha.1`, however I couldn't reproduce. using https://github.com/web-infra-dev/rspack-repro. Are you able to reproduce it?

**Comment by escaton** [comment]
@markmssd unfortunately no. But I hope having a clear example and a subset of changes that led to bug should be enough.

**Comment by chenjiahan** [comment]
This should related to: https://github.com/web-infra-dev/rspack/issues/7049

@ahabhgk cc

**Comment by ahabhgk** [comment]
Fix in 1.0.0-alpha.2
